pkg/webhook/pod_webhook.go

```diff
@@ -17,62 +17,219 @@ limitations under the License.
+		framework := getFrameworkForWebhook(wh.Name, "v")
+		if enabledFrameworks[framework] {
+			newWebhook.Webhooks = append(newWebhook.Webhooks, wh)
 		}
 	}
-	return newWebHook
 
+	mergeNamespaceSelectors(newWebhook, podSelector)
+	return newWebhook
 }
 
 func ModifyPodBasedMutatingWebhook(kueueCfg kueue.KueueConfiguration, currentWebhook *admissionregistrationv1.MutatingWebhookConfiguration) *admissionregistrationv1.MutatingWebhookConfiguration {
-	// if there is a pod based webhook, we need to safely enabled it
-	// For now we will not modify this
-	for _, val := range kueueCfg.Integrations.Frameworks {
-		if val == kueue.KueueIntegrationPod || val == kueue.KueueIntegrationDeployment || val == kueue.KueueIntegrationStatefulSet {
-			return currentWebhook
+	newWebhook := currentWebhook.DeepCopy()
+	newWebhook.Webhooks = []admissionregistrationv1.MutatingWebhook{}
+
+	enabledFrameworks := make(map[string]bool)
+	podSelector := defaultLabelSelector()
+
+	for _, fw := range kueueCfg.Integrations.Frameworks {
+		enabledFrameworks[string(fw)] = true
+	}
+
+	for _, wh := range currentWebhook.Webhooks {
+		framework := getFrameworkForWebhook(wh.Name, "m")
+		if enabledFrameworks[framework] {
+			newWebhook.Webhooks = append(newWebhook.Webhooks, wh)
+		}
+	}
+
+	mergeNamespaceSelectors(newWebhook, podSelector)
+	return newWebhook
+}
+
+func isPodBased(framework string) bool {
+	switch framework {
+	case "pod", "deployment", "statefulset":
+		return true
+	}
+	return false
+}
+
+func isClusterScopedFramework(framework string) bool {
+	switch framework {
+	case "kueue.x-k8s.io/clusterqueue", "kueue.x-k8s.io/resourceflavor", "kueue.x-k8s.io/cohort":
+		return true
+	}
+	return false
+}
+
+func getFrameworkForWebhook(name, prefix string) string {
+	suffix := strings.TrimSuffix(strings.TrimPrefix(name, prefix), ".kb.io")
+
+	for framework, s := range webhooksList {
+		if s == suffix {
+			return framework
 		}
 	}
-	newWebHook := currentWebhook.DeepCopy()
-	newWebHook.Webhooks = []admissionregistrationv1.MutatingWebhook{}
+	panic(fmt.Sprintf("unregistered framework: %s", name))
+}
 
-	for _, val := range currentWebhook.Webhooks {
-		if !findWebhook(val.Name, podBasedMutatingWebhooks) {
-			newWebHook.Webhooks = append(newWebHook.Webhooks, val)
+// mergeNamespaceSelectors applies merged namespace selectors to all webhooks in the configuration.
+// Example:
+//
+//		Existing webhook selector:
+//		namespaceSelector:
+//		  matchExpressions:
+//	   		- key: kubernetes.io/metadata.name
+//			  operator: NotIn
+//			  values:
+//			  - kube-system
+//			  - kueue-system
+//
+//		Pod integration selector:
+//		  matchExpressions:
+//		    - key: kueue.openshift.io/managed
+//		      operator: In
+//		      values: [true]
+//
+//		Merged selector:
+//		  matchLabels:
+//		    environment: prod
+//		  matchExpressions:
+//		    - key: kueue.openshift.io/managed
+//		      operator: In
+//		      values: [true]
+//		    - key: kubernetes.io/metadata.name
```

Comment:
I think this is a typo... metadata.namespace

Reply:
This line came from: https://github.com/openshift/kueue-operator/blob/main/bindata/assets/kueue-operator/validatingwebhook.yaml#L21

Reply:
Ok. metadata.name does correlate to the namespace.